Test helper

Deardrops opened the pull request:
TODOs:

- [x] improve *helper action* function
- [x] refactor testHelper's logic
- [x] set language in server page
- [x] log user action and show in textarea
- [x] disable *helper action* button until start battle
- [x] ~~localize page~~
- [x] style page
- [ ] write test wiki article
- [ ] compatible with firefox ?

Deardrops committed:
test-helper: click button take over input in console

1. disable BGM defaultly
2. get global variable "game" by click "updateBattle"
3. "upgrade" - upgrade your lrig to max level && put all lrig card to
lrigZone
4. add "card-name" input box for add card action

Deardrops committed:
one-key oben game && select deck in Server page

get Deck file names & Deck by localStorage
lower-case spell the button text
add placeholder to card name input

Deardrops committed:
check code style

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,222 @@
+  if (sockets.length !== 2) {
+    console.log("two client needed");
+    return;
+  };
```
😓

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,222 @@
+  this.deckName = '';
+
+  this.cfg = {
+    disableAudio: true,
```
😓

Deardrops committed:
improve TestHelper

click Oben! == open 2 client, create 1 room, start battle with selected
deck
2 player can play with 2 separate deck
show deckEditor by iframe in server page
keep previous test html and js

Deardrops committed:
improve test helper

improve helper function
set language in serve page
log action and state in textarea

Deardrops committed:
improve test helper

style page
hide helper action button until start battle

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,234 @@
+      lrigCards.push(card);
+    }
+  });
+  game.moveCards(lrigCards, p.lrigZone);
```
怎么保证最大等级在上面的

应该是移动最大等级，其他的要加`{bottom: true}`，否则触发出场效果

Deardrops replied:
已修改逻辑：
先找出最高等级的分身，放到分身区上方（可选择发动效果），
再将所有其他分身放置到分身区下方（不发动任何效果）。

塔8有两张5级分身，取前一张升级。这里不让用户选择要升级的分身。

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,234 @@
+function log(text) {
+  var logger = $('log');
+  logger.innerHTML += text;
+  logger.innerHTML += '\n';
```
用 textContent 吧，看到 innerHtml 总想到 XSS

Deardrops committed:
improve test helper

refactor test helper
grow lrig to max level and put other lrig card to the bottom of lrig
zone
aviod use innerHTML

webxoss commented on index.html:
```diff
@@ -0,0 +1,341 @@
+  -webkit-appearance: none;
+  /* Safari 和 Chrome */
+  border: none;
+  width: 7em;
```
这样式好丑啊。。。

webxoss commented on index.html:
```diff
@@ -0,0 +1,341 @@
+button:disabled,
+button[disabled] {
+  border: none;
+  filter: alpha(opacity=40);
```
去掉

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,277 @@
+    location.reload();
+    return;
+  }
+  var win = window.open('../webxoss-client/?local=true');
```
`./`

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,277 @@
+  }
+  var win = window.open('../webxoss-client/?local=true');
+  win.addEventListener('load', function() {
+    var win2 = window.open('../webxoss-client/?local=true');
```
`./?local=true`

webxoss commented on testHelper.js:
```diff
@@ -0,0 +1,277 @@
+    }
+  });
+  lrigCards.pop().moveTo(p.lrigZone);
+  lrigCards.forEach(function(card) {
```
`game.moveCards`

webxoss commented on index.html:
```diff
@@ -0,0 +1,341 @@
+          <span>to</span>
+          <select id="target-player" class="target-player">
+            <option value="">yourself&#39;s</option>
+            <option value="opponent">opponent&#39;s</option>
```
turn player's

webxoss commented on index.html:
```diff
@@ -0,0 +1,341 @@
+        </select>
+      </div>
+    </div>
+    <iframe id="deck-editor" src="../webxoss-client/DeckEditor/" scrolling="no" frameborder="no" border="0"></iframe>
```
`./`

webxoss commented on index.html:
```diff
@@ -0,0 +1,341 @@
+        <button onclick="resetLrigDeck();">reset lrig deck</button>
+      </div>
+      <div class="flex-center">
+        <textarea id="log"></textarea>
```
readonly

webxoss commented:
自动不换牌好弄吗？

Deardrops committed:
review
